api/v1beta1/cluster_types.go

```diff
@@ -143,6 +143,11 @@ type MachineDeploymentTopology struct {
 	// the values are hashed together.
 	Name string `json:"name"`
 
+	// FailureDomain is the failure domain the machines will be created in.
+	// Must match a key in the FailureDomains map stored on the cluster object.
```

Comment:
I'll open an issue.
This is another use case where we are putting changes on hold (like for upgrades), so we should make sure that the TopologyReconciled condition is updated accordingly
@ykakarap PTAL

Reply:
Noted. Looks like that work will come in a follow up PR.
To help with my understanding, we are looking to update the `FailureDomain` on the MD only after the cluster object is reconciled and `cluster.status.failureDomains` is available? 
OR
Are we planing to hold creating the MD till `cluster.status.failureDomains` is available so that we can validate the provided failureDomain and use that to create the MD.?

Reply:
@ykakarap The second option sounds better to me.